src/Storages/StorageFile.cpp

```diff
@@ -115,8 +115,12 @@ void listFilesWithRegexpMatchingImpl(
     {
         try
         {
-            fs::path path = fs::canonical(path_for_ls + for_match);
-            result.push_back(path.string());
+            /// We use fs::canonical to check if the file exists but the result path
+            /// will be fs::absolute.
+            /// Otherwise it will not allow to work with symlinks in `user_files_path` directory.
+            fs::path canonical_path = fs::canonical(path_for_ls + for_match);
```

Comment:
Is `canonical_path` used anywhere besides its declaration?

Reply:
It's only used here, for resolving path and check if the path does exist

Reply:
Why declare and define a standalone variable for it than if it is not used later? Why not just call `fs::canonical(...)`?

Reply:
@zvonand I've removed it.